code/game/objects/items/toys.dm

```diff
@@ -704,6 +706,58 @@
 	user.put_in_hands(H)
 	user.visible_message("[user] draws a card from the deck.", "<span class='notice'>You draw a card from the deck.</span>")
 	update_icon()
+	update_desc()
+
+obj/item/toy/cards/deck/AltClick(mob/user)
+	if(isliving(user))
+		var/mob/living/L = user
+		if(!(L.mobility_flags & MOBILITY_PICKUP))
+			return
+		if(cards.len == 0)
+			to_chat(user, "<span class='warning'>There are no more cards to draw!</span>")
+			return
+		else
+			var/drawsize = input(user, "How many cards to draw? (1-[min(cards.len,10)])", "Cards") as null|num
+			if (drawsize)
+				drawsize=clamp(drawsize,1,min(cards.len,10))
+				if(drawsize==1)
+					draw_card(user)//if drawing a single card, no need to use the many proc
+				else
+					draw_many(drawsize,user)//this one draws many cards at once
+	return
+
+obj/item/toy/cards/deck/proc/draw_many(dr,mob/user) //Number of cards to draw, mob who is drawing them
+	var/choice = null
+	var/obj/item/toy/cards/cardhand/C = new/obj/item/toy/cards/cardhand(user.loc)
+	var/obj/item/toy/cards/singlecard/H = new/obj/item/toy/cards/singlecard(user.loc)
+	var/i
+	var/O = src
+	for (i=1,i<=dr,i++)
```

Comment:
A lot of this is cargo-cult copied from the other drawcard. I'm not convinced half of it does anything. But I don't want to leave out something and all of a sudden the holodeck cards become real or some shit....basically I need a little helpsies with this "draw multiple" part >.> it works as intended testing but....